Makefile

```diff
@@ -14,10 +18,23 @@ BASE_IMAGE = $(shell grep FROM Dockerfile | grep -vi ' AS ' | awk '{print $$2}'
 build:
 	# update base image to get latest changes
 	docker pull ${BASE_IMAGE}
-	docker build -t $(IMAGE) --build-arg VERSION=$(VERSION) .
-	@echo Build $(IMAGE) successful
-	@mkdir -p bin
-	@echo $(IMAGE) > bin/latest_image
+	REPO=$(REPO) IMAGE_NAME=$(NAME) TAG=$(VERSION) VERSION=$(VERSION) bash scripts/package
+
+.PHONY: buildx-machine
+buildx-machine:
+	docker buildx ls | grep $(MACHINE) >/dev/null || \
+		docker buildx create --name=$(MACHINE) --platform=$(DEFAULT_PLATFORMS)
+
+# variables set by GHA workflow rancher/ecm-distro-tools/actions/publish-image:
+# - REPO: image repo, include $registry/$repo_path
+# - TAG: image tag
+# - TARGET_PLATFORMS: to be passed for buildx's --platform option
+# - IID_FILE_FLAG: options to generate image ID file
+.PHONY: workflow-image-build-push workflow-image-build-push-secure
+workflow-image-build-push: buildx-machine
+	MACHINE=$(MACHINE) OUTPUT_ARGS='--push' VERSION=$(VERSION) bash scripts/package
+workflow-image-build-push-secure: buildx-machine
+	MACHINE=$(MACHINE) OUTPUT_ARGS='--push' VERSION=$(VERSION) IS_SECURE=true bash scripts/package
```

Comment:
Cam we allow engineers to use `make IS_SECURE=true run` to test the production build locally?

Reply:
We still provide `build` target for local development, and the flag can be passed through envvar, e.g.:

```
export IS_SECURE=true
make build
```

Reply:
Tested that we can retrieve `IS_SECURE` from the environment and use it in the codebase.